ctru-rs/examples/file-explorer.rs

```diff
@@ -85,34 +115,49 @@ impl<'a> FileExplorer<'a> {
                     println!("{:2} - {}", i, entry.file_name().to_string_lossy());
                     self.entries.push(entry);
 
-                    // Paginate the output
                     if (i + 1) % 20 == 0 {
-                        println!("Press A to go to next page, or Start to exit");
-
-                        while self.apt.main_loop() {
-                            self.hid.scan_input();
-                            let input = self.hid.keys_down();
-
-                            if input.contains(KeyPad::KEY_A) {
-                                break;
-                            }
-
-                            if input.contains(KeyPad::KEY_START) {
-                                self.running = false;
-                                return;
-                            }
-
-                            self.gfx.wait_for_vblank();
-                        }
+                        self.wait_for_page_down();
                     }
                 }
                 Err(e) => {
                     println!("{} - Error: {}", i, e);
                 }
             }
         }
+    }
 
-        println!("Start to exit, A to select an entry by number, B to go up a directory, X to set the path.");
+    fn print_file_contents(&mut self) {
+        match std::fs::read_to_string(&self.path) {
+            Ok(contents) => {
+                println!("File contents:\n{0:->80}", "");
+                println!("{contents}");
```

Comment:
Maybe we should paginate the output here too? I wonder if we should move the pagination message to the lower screen to avoid interrupting the top screen's output.

Reply:
Oops, good point, I think when I had BufReader here it made more sense but using `read_to_string` is probably questionable when trying to paginate. Using the bottom screen to prompt for it probably takes a little work but sounds like a potentially nicer UX for this example.

Reply:
I think the prompt move should be as making a new temporary console and selecting it.